crates/oxc_semantic/src/builder.rs

```diff
@@ -189,7 +189,7 @@ impl<'a> SemanticBuilder<'a> {
     /// # Panics
     pub fn build(mut self, program: &Program<'a>) -> SemanticBuilderReturn<'a> {
         if self.source_type.is_typescript_definition() {
-            let scope_id = self.scope.add_scope(None, ScopeFlags::Top);
+            let scope_id = self.scope.add_scope(None, AstNodeId::dummy(), ScopeFlags::Top);
```

Comment:
Why not `self.current_node_id` here?

Reply:
Good question. I guess we could do that.

Reply:
Please feel free to submit a PR.

Reply:
We don't build semantic for `.d.ts` file now. So we haven't  current_node_id. We may need to build semantic for `d.ts` in the future. Rolldown needs this to bundle types to one entry point.